tgui/packages/tgui/interfaces/ApcControl.js

```diff
@@ -0,0 +1,300 @@
+          <Window.Content scrollable>
+            <LogPanel />
+          </Window.Content>
+        </Box>
+      )}
+    </Fragment>
+  );
+};
+
+const ControlPanel = (props, context) => {
+  const { act, data } = useBackend(context);
+  const {
+    emagged,
+    logging,
+  } = data;
+  const [
+    sortByField,
+    setSortByField,
+  ] = useLocalState(context, 'sortByField', null);
+  return (
+    <Flex>
+      <Flex.Item>
+        <Box inline mr={2} color="label">
+          Sort by:
+        </Box>
+        <Button.Checkbox
+          checked={sortByField === 'name'}
+          content="Name"
+          onClick={() => setSortByField(sortByField !== 'name' && 'name')} />
+        <Button.Checkbox
+          checked={sortByField === 'charge'}
+          content="Charge"
+          onClick={() => setSortByField(
+            sortByField !== 'charge' && 'charge'
+          )} />
+        <Button.Checkbox
+          checked={sortByField === 'draw'}
+          content="Draw"
+          onClick={() => setSortByField(sortByField !== 'draw' && 'draw')} />
+      </Flex.Item>
+      <Flex.Item grow={1} />
+      <Flex.Item>
+        {emagged === 1 && (
+          <Fragment>
+            <Button
+              color={logging === 1 ? 'bad' : 'good'}
+              content={logging === 1 ? 'Stop Logging' : 'Restore Logging'}
+              onClick={() => act('toggle-logs')}
+            />
+            <Button
+              content="Reset Console"
+              onClick={() => act('restore-console')}
+            />
+          </Fragment>
+        )}
+        <Button
+          color="bad"
+          content="Log Out"
+          onClick={() => act('log-out')}
+        />
+      </Flex.Item>
+    </Flex>
+  );
+};
+
+const ApcControlScene = (props, context) => {
+  const { data, act } = useBackend(context);
+
+  const [
+    sortByField,
+  ] = useLocalState(context, 'sortByField', null);
+
+  const apcs = flow([
+    map((apc, i) => ({
+      ...apc,
+      // Generate a unique id
+      id: apc.name + i,
+    })),
+    sortByField === 'name' && sortBy(apc => apc.name),
+    sortByField === 'charge' && sortBy(apc => -apc.charge),
+    sortByField === 'draw' && sortBy(
+      apc => -powerRank(apc.load),
+      apc => -parseFloat(apc.load)),
+  ])(data.apcs);
+  return (
+    <Table>
+      <Table.Row header>
+        <Table.Cell>
+          On/Off
+        </Table.Cell>
+        <Table.Cell>
+          Area
+        </Table.Cell>
+        <Table.Cell collapsing>
+          Charge
+        </Table.Cell>
+        <Table.Cell collapsing textAlign="right">
+          Draw
+        </Table.Cell>
+        <Table.Cell collapsing title="Equipment">
+          Eqp
+        </Table.Cell>
+        <Table.Cell collapsing title="Lighting">
+          Lgt
+        </Table.Cell>
+        <Table.Cell collapsing title="Environment">
+          Env
+        </Table.Cell>
+      </Table.Row>
+      {apcs.map((apc, i) => (
+        <tr header
```

Comment:
remove header prop, also missing Table__row class (although it won't do much, i think).

Reply:
Gottcha